admin/class-aioseop-notices.php

```diff
@@ -0,0 +1,711 @@
+		 * @access private
+		 *
+		 * @see self::notices
+		 * @see self::active_notices
+		 */
+		private function obj_load_options() {
+			$notices_options = $this->obj_get_options();
+
+			$this->notices        = $notices_options['notices'];
+			$this->active_notices = $notices_options['active_notices'];
+		}
+
+		/**
+		 * Get AIOSEOP_Notice Options
+		 *
+		 * @since 3.0
+		 * @access private
+		 *
+		 * @return array
+		 */
+		private function obj_get_options() {
+			$defaults = array(
+				'notices'        => array(),
+				'active_notices' => array(),
+			);
+
+			$notices_options = get_option( 'aioseop_notices' );
+			if ( false === $notices_options ) {
+				return $defaults;
+			}
+
+			return wp_parse_args( $notices_options, $defaults );
+		}
+
+		/**
+		 * Update Notice Options
+		 *
+		 * @since 3.0
+		 * @access private
+		 *
+		 * @return boolean True if successful, using update_option() return value.
+		 */
+		private function obj_update_options() {
+			$notices_options     = array(
+				'notices'        => $this->notices,
+				'active_notices' => $this->active_notices,
+			);
+			$old_notices_options = $this->obj_get_options();
+			$notices_options     = wp_parse_args( $notices_options, $old_notices_options );
+
+			return update_option( 'aioseop_notices', $notices_options );
+		}
+
+		/**
+		 * Notice Default Values
+		 *
+		 * Returns the default value for a variable to be used in self::notices[].
+		 *
+		 * @since 3.0
+		 *
+		 * @see self::notices Array variable that stores the collection of notices.
+		 *
+		 * @return array Notice variable in self::notices.
+		 */
+		public function notice_defaults() {
+			return array(
+				'slug'           => '',
+				'delay_time'     => 0,
+				'message'        => '',
+				'action_options' => array(),
+				'class'          => 'notice-info',
+				'target'         => 'site',
+				'screens'        => array(),
+				'time_start'     => time(),
+				'time_set'       => time(),
+			);
+		}
+
+		/**
+		 * Action Options Default Values
+		 *
+		 * Returns the default value for action_options in self::notices[$slug]['action_options'].
+		 *
+		 * @since 3.0
+		 *
+		 * @return array Action_Options variable in self::notices[$slug]['action_options'].
+		 */
+		public function action_options_defaults() {
+			return array(
+				'time'    => 0,
+				'text'    => __( 'Dismiss', 'all-in-one-seo-pack' ),
+				'link'    => '#',
+				'dismiss' => true,
+				'class'   => '',
+			);
+		}
+
+		/**
+		 * Set Notice Action Options
+		 *
+		 * Sets the Action Options in a Notice.
+		 *
+		 * @since 3.0
+		 * @access private
+		 *
+		 * @see self::insert_notice()
+		 * @see self::update_notice()
+		 *
+		 * @param array $action_options New action options to be added/updated.
+		 * @return array Action Options with new values added to old.
+		 */
+		private function set_action_options( $action_options ) {
+			$rtn_action_options = array();
+			if ( empty( $action_options ) && ! is_array( $action_options ) ) {
```

Comment:
This seems confusing. should this be OR instead of AND?

Reply:
Nice catch! ...and I was thinking the same. Although, it is intended to skip this operation when a notice has no action(s), and to have some form of default to a dismiss button.

I think it's `empty( $action_options )` that needs to be removed. I also added a note since it did seem rather confusing.